internal/storage/s3backend/client_test.go

```
@@ -0,0 +1,279 @@
+	cfg := &s3backend.Config{
+		Bucket:             "test-bucket",
+		Region:             "ap-south-1",
+		AWSAccessKeyID:     "AKIAEXAMPLE",
+		AWSSecretAccessKey: "secretexample",
+	}
+
+	s, err := s3backend.New(context.Background(), cfg, logger.NewNoopLogger())
+	require.NoError(t, err)
+	require.NotNil(t, s)
+
+	var _ storage.Storage = s
```

golangci-lint (2.12.2)

[error] QF1011: could omit type storage.Storage from declaration; it will be inferred from the right-hand side

(staticcheck)